localstack/utils/analytics/response_aggregator.py

```diff
@@ -0,0 +1,74 @@
+import atexit
+import datetime
+import logging
+import threading
+from collections import Counter, namedtuple
+from typing import Any, Dict
+
+from localstack import config
+from localstack.utils import analytics
+from localstack.utils.scheduler import Scheduler
+
+LOG = logging.getLogger(__name__)
+FLUSH_INTERVAL_SECS = 10
+
+
+ResponseInfo = namedtuple("ResponseInfo", "service, operation, status_code")
+
+
+class ResponseAggregator:
+    """
+    Collects HTTP response data, aggregates it into small batches, and periodically emits (flushes) it as an analytics event
+    """
+
+    def __init__(self):
+        self.response_counter = Counter()
+        self.period_start_time = datetime.datetime.utcnow()
+        self.flush_scheduler = None
+
+    def start_thread(self):
+        """
+        Start a thread that periodically flushes HTTP response data aggregations as analytics events
+        """
+        self.flush_scheduler = Scheduler()
+        scheduler_thread = threading.Thread(target=self.flush_scheduler.run)
```

Comment:
we should start the thread as a daemon thread so it is terminated with the interpreter.